source/includes/connection-snippets/x509-snippets/python-connection.py

```diff
@@ -0,0 +1,14 @@
+# begin x509 connection
+from pymongo import MongoClient
+
+uri = "mongodb+srv://<cluster-url>/?authMechanism=MONGODB-X509"
+client = MongoClient(uri,
+                     tls=True,
```

Comment:
I don't think this is necessary, as PyMongo should enable TLS by virtue of `mongodb+srv` being used (per [driver spec](https://github.com/mongodb/specifications/blob/master/source/initial-dns-seedlist-discovery/initial-dns-seedlist-discovery.rst#seedlist-discovery)). The [unified URI options spec](https://github.com/mongodb/specifications/blob/master/source/uri-options/uri-options.rst) also suggests enabling TLS if one or more `tls*` options are used (e.g. `tlsCertificateKeyFile` below), although I'd defer to @ShaneHarvey and @prashantmital to confirm whether PyMongo honors that.

Reply:
`tls=True` should technically not be required, but it's needed for this snippet. I actually need to update all the snippets to have `tls=True` in all the URI's because some drivers currently don't work without it.

Reply:
Can you clarify which drivers require `tls=true` to be explicitly specified? If any driver is not complying with the [Initial DNS Seedlist Discovery](https://github.com/mongodb/specifications/blob/master/source/initial-dns-seedlist-discovery/initial-dns-seedlist-discovery.rst) spec requirement to enable TLS by default when the `mongodb+srv` scheme is used that would be something we should fix immediately.

Reply:
I don't have a full list of which drivers require tls=true to be explicitly specified. I can create a list of drivers that don't follow the spec exactly.  I do know for sure that Go requires it to be explicitly specified and throws an error without it (I think there's at least one other driver that does to, but I'd have to test it out).

Reply:
I can confirm that PyMongo will automatically default to `tls=true` when `mongodb+srv://` is used.

Reply:
For this PR we're sticking to `tls=true` because other drivers require it. I have another ticket (https://jira.mongodb.org/browse/DOCSP-7932) for updating these snippets once all the drivers match the spec, but currently, since some drivers require it, this bit has to be there.